src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts

```diff
@@ -113,32 +108,13 @@ export class DataFieldEditDialogComponent {
         // an id means this data option is being edited
         if (dataOption.id) {
           // update the displayed data options
-          const existingIndex = this.data.dataFieldOptions.findIndex(x => x.id === dataOption.id);
-          this.data.dataFieldOptions[existingIndex] = dataOption;
-          // make sure this data option gets saved when we are finished
-          const changedIndex = this.changedDataFieldOptions.findIndex(x => x.id === dataOption.id);
-          // is it on the changed list already?
-          if (changedIndex < 0) {
-            // not on changed list, is it on the added list?
-            const addedIndex = this.addedDataFieldOptions.findIndex(ado => ado.id === dataOption.id);
-            if (addedIndex < 0) {
-              // not on added list either, so add to changed list
-              this.changedDataFieldOptions.push(dataOption);
-            } else {
-              // on added list, so update with new values
-              this.addedDataFieldOptions[addedIndex] = dataOption;
-            }
-          } else {
-            // on the changed list, so update with new values
-            this.changedDataFieldOptions[changedIndex] = dataOption;
-          }
+          const existingIndex = this.data.dataField.dataOptions.findIndex(x => x.id === dataOption.id);
+          this.data.dataField.dataOptions[existingIndex] = dataOption;
         } else {
           // no id means this data option has just been entered
           dataOption.id = uuidv4();
           // add to displayed data options
-          this.data.dataFieldOptions.push(dataOption);
-          // make sure data option gets added when we are done
-          this.addedDataFieldOptions.push(dataOption);
+          this.data.dataField.dataOptions.push(dataOption);
```

Comment:
Do we have to worry about displayOrder here?

Reply:
Per my comment above, go ahead and add a Jira ticket for that.